sktime/forecasting/compose/_pipeline.py

```diff
@@ -1316,6 +1335,13 @@ def __init__(
         self.fh_X = fh_X
         self.behaviour = behaviour
         self.columns = columns
+        self.forecaster_X_exogeneous = forecaster_X_exogeneous
+        if isinstance(forecaster_X_exogeneous, str):
+            if forecaster_X_exogeneous not in ["None", "complement"]:
+                raise ValueError(
+                    'forecaster_X_exogeneous must be one of "None", "complement",'
+                    "or a pandas.Index coercible"
+                )
```

Comment:
I think we also need to check in case `forecaster_X_exogeneous` is empty list `[]`. It's probably going to fail then.

Probably applicable for `columns` as well.